packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -1527,4 +1527,92 @@ void main() {
     expect((mergeableMaterial.children.first as MaterialSlice).color, firstPanelColor);
     expect((mergeableMaterial.children.last as MaterialSlice).color, secondPanelColor);
   });
+
+  testWidgets('ExpansionPanel.iconColor, ExpansionPanel.expandedIconColor, ExpansionPanel.disbaledColor test', (WidgetTester tester) async {
+    const Color firstIconColor = Colors.red;
+    const Color expandedIconColor = Colors.blue;
+    const Color disabledIconColor = Colors.amber;
+
+    await tester.pumpWidget(MaterialApp(
+      home: SingleChildScrollView(
+        child: ExpansionPanelList(
+          children: <ExpansionPanel>[
+            ExpansionPanel(
+              headerBuilder: (BuildContext context, bool isExpanded) {
+                return const Text('A');
+              },
+              body: const SizedBox(height: 100.0),
+              iconColor: MaterialStateProperty.resolveWith((Set<MaterialState>states) {
+                if (states.contains(MaterialState.selected)) {
+                  return expandedIconColor;
+                } else if (states.contains(MaterialState.disabled)) {
+                  return disabledIconColor;
+                }
+
+                return firstIconColor;
+
+              })
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    ExpandIcon expandIcon = tester.widget(find.byType(ExpandIcon));
+
+    expect(expandIcon.color, firstIconColor);
+
+    await tester.tap(find.byType(ExpandIcon));
+    await tester.pump(const Duration(milliseconds: 200));
+    await tester.pumpAndSettle();
+
+    expandIcon = tester.widget(find.byType(ExpandIcon));
+
+    expect(expandIcon.expandedColor, expandedIconColor);
+    expect(expandIcon.isExpanded, true);
```

Comment:
I'm not sure why but the tests are failing. Apparently `isExpanded` is always false even after `tester.tap()` is executed. 

Shouldn't it be true after we tap on the expandIcon? 

I tested the feature manually and it seems to be working fine by creating an example to test it.

Reply:
As far as I can tell, there are two issues. First, [this conditional](https://github.com/flutter/flutter/blob/38840b41a9414607156c33296596fd2f2e02ba0a/packages/flutter/lib/src/material/expansion_panel.dart#L441) is resolving to false, meaning the `setState` below is never reached.

Second, even if that resolved to true, [the very next line](https://github.com/flutter/flutter/blob/38840b41a9414607156c33296596fd2f2e02ba0a/packages/flutter/lib/src/material/expansion_panel.dart#L442) would throw an exception because it attempts to cast an `ExpansionPanel` as an `ExpansionPanelRadio`.

I think the next step is to investigate those two issues. Your handler is getting called, so we at least don't have to debug lost tap events :)